hydroflow/examples/two_pc/coordinator.rs

```diff
@@ -0,0 +1,232 @@
+            subordinates_out_tee_1_push,
+            hf.start_tee()
+                .tee(subordinates_out_tee_2_push, subordinates_out_tee_3_push),
+        ),
+    );
+
+    // Demultiplex received messages into phase1 and phase2 handoffs
+    let (p1_recv_push, p1_recv_pull) = hf.make_edge::<_, VecHandoff<SubordResponse>, _>("p1");
+    let (p2_recv_push, p2_recv_pull) = hf.make_edge::<_, VecHandoff<SubordResponse>, _>("p2");
+
+    // Create a flow to partition the received messages and populate the handoffs
+    hf.add_subgraph(
+        "demux",
+        msg_recv.flatten().pull_to_push().partition(
+            |m| m.mtype == MsgType::Commit || m.mtype == MsgType::Abort,
+            hf.start_tee().map(Some).push_to(p1_recv_push),
+            hf.start_tee().map(Some).push_to(p2_recv_push),
+        ),
+    );
+
+    // Multiplex messages to send
+    let (p1_send_push, p1_send_pull) =
+        hf.make_edge::<_, VecHandoff<(String, CoordMsg)>, _>("p1 send");
+    let (p2_send_push, p2_send_pull) =
+        hf.make_edge::<_, VecHandoff<(String, CoordMsg)>, _>("p2 send");
+    let (p2_send_end_push, p2_send_end_pull) =
+        hf.make_edge::<_, VecHandoff<(String, CoordMsg)>, _>("p2 send end");
+
+    // Create a flow to union together the messages and populate the network with them
+    hf.add_subgraph(
+        "mux",
+        p1_send_pull
+            .chain(p2_send_pull)
+            .chain(p2_send_end_pull)
+            .pull_to_push()
+            .push_to(msg_send),
+    );
+
+    // Phase 1 request flow:
+    // Given a transaction commit request from stdio, send a Prepare Message to each subordinate
+    // with the transaction ID and a unique message ID.
+    // HACK: We move the xids HashSet into a map operator as flow state.
+    // Should be done with a groupby.
+    let mut xids = HashSet::new();
+    hf.add_subgraph(
+        "phase 1 init",
+        text_out
+            .flatten()
+            .map(move |xidstr| match xidstr.trim().parse::<u16>() {
+                Ok(the_xid) => {
+                    let first_use = xids.get(&the_xid);
+                    match first_use {
+                        None => {
+                            xids.insert(the_xid);
+                            Some(CoordMsg {
+                                xid: the_xid,
+                                mid: 1, // first message for this transaction
+                                mtype: MsgType::Prepare,
+                            })
+                        }
+                        _ => {
+                            println!("Transaction ID {} already used", the_xid);
+                            None
+                        }
+                    }
+                }
+                Err(_) => None,
+            })
+            .cross_join(subordinates_out_tee_1_pull.flatten())
+            .map(|(msg, addr)| match msg {
+                None => None,
+                _ => Some((addr, msg.unwrap())),
+            })
```

Comment:
Will just be different after fixing above. Will need a new map to `Some(...)` for the `push_to`

Reply:
LGTM